lib/sqlalchemy/sql/functions.py

```diff
@@ -49,7 +53,18 @@ def register_function(identifier, fn, package="_default"):
 
     """
     reg = _registry[package]
+    if identifier in reg:
```

Comment:
```python
if not case_sensitive:
    identifier = identifier.lower()
```
could be a simpler approach?

Reply:
I want to be able to have an identifier with any case, even if it is marked as case insensitive. For example, the `ST_TrAnSfOrm()` function should be compiled as `ST_Transform()`, not as `st_transform()`.